include/Parser/Parser.hpp

```diff
@@ -1,5 +1,36 @@
+#ifndef GLU_PARSER_HPP
+#define GLU_PARSER_HPP
+
+#include "Basic/Tokens.hpp"
+#include "Lexer/Scanner.hpp"
+
+namespace glu {
+class BisonParser; // Forward declaration of the class generated by Bison
+}
+
+#include "Parser.tab.hpp" // Include the generated Bison parser
+
+namespace glu {
+
 class Parser {
+    Scanner &scanner;
+    std::unique_ptr<BisonParser> parser;
```

Comment:
you don't need a unique_ptr, BisonParser is fully defined in "Parser.tab.hpp" which is included